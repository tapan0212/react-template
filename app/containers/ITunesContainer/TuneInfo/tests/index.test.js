/**
 *
 * Tests for TuneInfo
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import { TuneContainerTest as TuneInfo } from '../index';

describe('<TuneInfo />', () => {
  const mockSongData = {
    trackId: 1,
    artistName: 'Ed',
    trackName: 'Perfect',
    releaseDate: '2017-05-10'
  };
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<TuneInfo selectedTune={mockSongData} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 TuneInfo component', () => {
    const { getAllByTestId } = renderWithIntl(<TuneInfo selectedTune={mockSongData} />);
    expect(getAllByTestId('tune-info').length).toBe(1);
  });
});
