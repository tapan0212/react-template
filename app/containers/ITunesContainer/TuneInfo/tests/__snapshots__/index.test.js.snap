// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TuneInfo /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      data-testid="tune-info"
    >
      <div
        class="TuneInfo__CustomInfoCard-sc-3vntfw-0 iwdbOp"
      >
        <div
          class="TuneInfo__CustomCard-sc-3vntfw-1 hfgazm"
        >
          <div
            class="TuneInfo__CustomImg-sc-3vntfw-2 KUsaJ"
          >
            <img
              alt=""
            />
          </div>
          <div>
            <p
              class="T__StyledText-gjlic1-0 cbAvsY"
              data-testid="t"
            >
              Tune: Perfect
            </p>
          </div>
          <div>
            <p
              class="T__StyledText-gjlic1-0 cbAvsY"
              data-testid="t"
            >
              Artist Name: Ed
            </p>
          </div>
          <div>
            <p
              class="T__StyledText-gjlic1-0 cbAvsY"
              data-testid="t"
            >
              Release Year: 2017
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`<TuneInfo /> should render and match the snapshot when there is no redux store 1`] = `
<body>
  <div>
    <div
      data-testid="tune-info"
    >
      <div
        class="TuneInfo__CustomInfoCard-sc-3vntfw-0 iwdbOp"
      >
        <div
          class="TuneInfo__CustomCard-sc-3vntfw-1 hfgazm"
        >
          <div
            class="TuneInfo__CustomImg-sc-3vntfw-2 KUsaJ"
          >
            <img
              alt=""
            />
          </div>
          <div>
            <p
              class="T__StyledText-gjlic1-0 cbAvsY"
              data-testid="t"
            >
              Tune: 
            </p>
          </div>
          <div>
            <p
              class="T__StyledText-gjlic1-0 cbAvsY"
              data-testid="t"
            >
              Artist Name: 
            </p>
          </div>
          <div>
            <p
              class="T__StyledText-gjlic1-0 cbAvsY"
              data-testid="t"
            >
              Release Year: 
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
